src/Orleans.Core/Messaging/ByteArrayBuilder.cs

```diff
@@ -125,32 +125,21 @@ private void EnsureRoomFor(int n)
         /// <returns></returns>
         public ByteArrayBuilder Append(byte[] array)
         {
+            // We shouldn't introduce external buffer Since we need return byte array to ArrayPool<byte>.Shared.
```

Comment:
@sergeybykov @jason-bragg I have no issue with removing the "large foreign buffer" optimization, at least for now. What do you think?

An alternative approach may be to explicitly mark which buffers are pooled and which aren't.

Reply:
I spoke with @jason-bragg and he is fine with this particular change